pkg/agent/agent.go

```diff
@@ -112,6 +123,41 @@ func (a *Agent) shutdown() {
 	}
 }
 
+func (a *Agent) setupProfiling() {
+	if a.c.ProfilingEnabled {
+		if runtime.MemProfileRate == 0 {
+			a.c.Log.Warn("Memory profiles are disabled")
+		}
+		if a.c.ProfilingPort > 0 {
+			grpc.EnableTracing = true
+			go func() {
+				port := strconv.Itoa(a.c.ProfilingPort)
+				a.c.Log.Info(http.ListenAndServe("localhost:"+port, nil))
+			}()
+		}
+		if a.c.ProfilingFreq > 0 {
+			c := &profiling.Config{
+				Tag:        "agent",
+				Frequency:  a.c.ProfilingFreq,
+				DebugLevel: 0,
+				Profiles:   []string{"goroutine", "threadcreate", "heap", "block", "mutex", "trace", "cpu"},
+			}
+			err := profiling.Start(c)
+			if err != nil {
+				a.c.Log.Error("Profiler failed to start: %v", err)
```

Comment:
a return here would avoid having to have an else clause, then we can call a.t.Go as the last statement in the method

Reply:
:100: